File: app/src/main/java/com/fphoenixcorneae/viewpagertransformer/DepthPageTransformer.java
package com.fphoenixcorneae.viewpagertransformer;

import android.view.View;

import com.nineoldandroids.view.ViewHelper;

/**
 * Created on 2017/6/30.
 */

public class DepthPageTransformer extends BaseTransformer {
    private static final float MIN_SCALE = 0.75F;

    @Override
    protected void onTransform(View page, float position) {
        if (position <= 0F) {
            ViewHelper.setTranslationX(page, 0F);
            ViewHelper.setScaleX(page, 1F);
            ViewHelper.setScaleY(page, 1F);
        } else if (position <= 1F) {
            float scaleFactor = MIN_SCALE + (1F - MIN_SCALE) * (1F - Math.abs(position));
            ViewHelper.setTranslationX(page, page.getWidth() * -position);
            ViewHelper.setScaleX(page, scaleFactor);
            ViewHelper.setScaleY(page, scaleFactor);
            ViewHelper.setPivotY(page, page.getHeight() * 0.5F);
            ViewHelper.setAlpha(page, 1F - position);
        }
    }

    @Override
    protected boolean isPagingEnabled() {
        return true;
    }
}

File: view-pager-transformer/src/main/java/com/fphoenixcorneae/view_pager_transformer/DepthPageTransformer.kt
package com.fphoenixcorneae.view_pager_transformer

import android.view.View
import com.nineoldandroids.view.ViewHelper
import kotlin.math.abs

/**
 * @desc 深度页面的转换效果
 * @date 2017-06-30
 */
class DepthPageTransformer : AbstractBaseTransformer() {
    override fun onTransform(page: View, position: Float) {
        if (position <= 0f) {
            ViewHelper.setTranslationX(page, 0f)
            ViewHelper.setScaleX(page, 1f)
            ViewHelper.setScaleY(page, 1f)
        } else if (position <= 1f) {
            val scaleFactor = MIN_SCALE + (1f - MIN_SCALE) * (1f - abs(position))
            ViewHelper.setTranslationX(page, page.width * -position)
            ViewHelper.setScaleX(page, scaleFactor)
            ViewHelper.setScaleY(page, scaleFactor)
            ViewHelper.setPivotY(page, page.height * 0.5f)
            ViewHelper.setAlpha(page, 1f - position)
        }
    }

    override val isPagingEnabled: Boolean = true

    companion object {
        private const val MIN_SCALE = 0.75f
    }
}